proparse/src/main/java/org/prorefactor/treeparser/TreeParserSuperSymbolScope.java

```diff
@@ -20,27 +22,27 @@
  * references to syntax tree nodes or child scopes. Is always generated either from a SymbolScopeRoot, or else from
  * another SymbolScopeSuper when a copy is being made.
  */
-public class SymbolScopeSuper extends SymbolScopeRoot {
+public class TreeParserSuperSymbolScope extends TreeParserRootSymbolScope {
 
   /**
    * TreeParser01 stores and looks up SymbolScopeSuper objects in this cache, which by default is a synchronizedMap
    * wrapped org.prorefactor.util.Cache object with a maximum cache size of 100. It is safe for any application to
    * completely override this. (Well, of course, be careful that you provide some mechanism for keeping the cache from
    * growing too large.) Since it's just a cache, it's completely exposed. Just don't make it null. :)
    */
-  public static Cache<String, SymbolScopeSuper> cache = CacheBuilder.newBuilder().maximumSize(100).build();
+  public static Cache<String, TreeParserSuperSymbolScope> cache = CacheBuilder.newBuilder().maximumSize(100).build();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this "public static cache" field final [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1444)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make cache a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)